environment.yml

```diff
@@ -10,3 +10,6 @@ dependencies:
   - pytest-mock
   - pytest-xdist
   - semver
+  - jupyter-lsp
+  - langchain
+  - notebook
```

Comment:
This does not seem like the right place to add these packages. 
```suggestion
```
I believe we should add them to the cpu.in and gpu.in files here:
https://github.com/aws/sagemaker-distribution/blob/main/build_artifacts/v0/v0.5/v0.5.1/cpu.env.in
https://github.com/aws/sagemaker-distribution/blob/main/build_artifacts/v0/v0.5/v0.5.1/gpu.env.in

Perhaps, best to check with one of the maintainers on the project. 

@balajisankar15 Can you help us adding new packages to the distribution, and guide on what other steps are needed? Thanks for your help.

Reply:
Yes, @3coins is right. 

Development process:
* Generate a new patch version locally (from the latest version).
* * ```python ./src/main.py create-patch-version-artifacts --base-patch-version 0.5.1```
* This will generate a separate build artifacts folder for the new version.
* Modify cpu.env.in and gpu.env.in in that folder to include your new packages.
* Add your Docker based test in the test artifacts folder . Ref: https://github.com/aws/sagemaker-distribution/blob/main/test/test_artifacts/autogluon.test.Dockerfile
* Modify https://github.com/aws/sagemaker-distribution/blob/main/test/test_dockerfile_based_harness.py to include your test docker file
* Now run the build command ``` python ./src/main.py build --target-patch-version 0.5.2```
* Just verify whether 
* * The script is able to generate a new version with your dependencies. For example , if the package you are trying to add depends on packages which are outdated, there is a high chance that the script will fail.
* * After we generate a new docker image based on the env.in files the script will try to run the tests. Please verify whether the tests related to your package are succeeding.

PR Process:
Now that you have verified that you are able to successfully generate a new patch version with your packages(also tests are succeeding), it is time to send out a PR.
* Just include your test docker file and any other supplementary scripts in the PR.
* Don't include the patch version (0.5.2) which you generated locally.
* Don't include the test_dockerfile_based_harness.py in which you referred your test docker file.
* Leave instructions in the PR description about what we should do to add your packages (specify the changes you made to env.in file).
* Once the PR is merged, we will include your new packages during the next minor version release.

Reply:
Example PR: https://github.com/aws/sagemaker-distribution/pull/44

Reply:
@balajisankar15 Thanks for the response! Is there something I need to run before the first `python ./src/main.py` command?

I get an error that ends with:

```
docker.errors.DockerException: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
```

I can share the entire ~100-line trace if it'll help.

Reply:
@JasonWeill 
That seems like a problem with your docker server. Can you make sure docker is running? On Mac OS, I have a Docker Desktop app running. Also, try running `docker info`.